### PR TITLE
[TRAFODION-3144] Correct Syntactic Descriptions of Character String Data Types and Add Examples (NCHAR) for *ALTER TABLE Statement* in *Trafodion SQL Reference Manual*

### PR DESCRIPTION
**Pull request by liuyu000**

### CODE REVIEW

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 1]
```diff
@@ -390,25 +390,27 @@ alter-action is:
 
 column-definition is:
    column-name data-type
-      ([DEFAULT default]
-         [[constraint constraint-name] column-constraint])
+      [[DEFAULT default]
+         [[constraint constraint-name] column-constraint]]
 
 data-type is:
      char[acter] [(length)[characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: This should say, "char[acter] (length [characters])"

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 2]
```diff
@@ -390,25 +390,27 @@ alter-action is:
...
+   | char[acter] varying (length [characters])
+                         [CHARACTER SET char-set-name]
+                         [UPSHIFT] [[NOT] CASESPECIFIC]
+   | varchar (length [characters]) 
+             [CHARACTER SET char-set-name]
+             [UPSHIFT] [[NOT] CASESPECIFIC]
+   | varchar2 (length [characters]) 
+              [CHARACTER SET char-set-name]
+              [UPSHIFT] [[NOT] CASESPECIFIC]         
+   | nchar [length [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: This should say, "nchar (length [characters])"

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 3]
```diff
@@ -756,6 +758,86 @@ Alter TABLE PRODUCT
 ALTER COLUMN vend_id RENAME TO cstm_id;
 ```
 
+* This example demonstrates the character set of NCHAR column cannot be altered.
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: See my comments on https://github.com/apache/trafodion/pull/1644.

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 2 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: `length [characters]` is optional.

default

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 1 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks for your review @DaveBirdsall 

`length [characters]` is optional.
If `length` is not specified, then default is 1 character.

default

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 3 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks @DaveBirdsall your comments have been incorporated 😁

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 2 above)
Comment (DaveBirdsall) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks; you are correct. So, it should say, "nchar [(length [characters])]". The length is optional, but if it is present, parentheses are required.

Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Yes, the origin text is right, I've corrected it 😁

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 4]
```diff
@@ -390,25 +390,27 @@ alter-action is:
...
-         [CHARACTER SET char-set-name]
-         [UPSHIFT] [[not] casespecific]
-   | char[acter] varying (length)
-         [character set char-set-name]
-         [upshift] [[not] casespecific]
-   | varchar (length) [character set char-set-name]
-         [upshift] [[not] casespecific]
-   | varchar2 (length) [character set char-set-name]
-         [upshift] [[not] casespecific]         
+     char[acter] [(length) [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Should be "char[acter] [(length [characters])]". The optional [characters] syntactic element should be inside the parenthesis.

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 5]
```diff
@@ -390,25 +390,27 @@ alter-action is:
...
+   | char[acter] varying (length [characters])
+                         [CHARACTER SET char-set-name]
+                         [UPSHIFT] [[NOT] CASESPECIFIC]
+   | varchar (length [characters]) 
+             [CHARACTER SET char-set-name]
+             [UPSHIFT] [[NOT] CASESPECIFIC]
+   | varchar2 (length [characters]) 
+              [CHARACTER SET char-set-name]
+              [UPSHIFT] [[NOT] CASESPECIFIC]         
+   | nchar [(length) [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Same here.

**Pull request merged**